persistence/src/main/scala/com/lightbend/lagom/internal/persistence/cluster/ClusterDistribution.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.persistence.cluster
+
+import java.net.URLEncoder
+import java.util.concurrent.TimeUnit
+
+import akka.actor.{ Actor, ActorRef, ActorSystem, DeadLetterSuppression, ExtendedActorSystem, Extension, ExtensionId, ExtensionIdProvider, Props, Terminated }
+import akka.cluster.sharding.ShardRegion.EntityId
+import akka.cluster.sharding.{ ClusterSharding, ClusterShardingSettings, ShardRegion }
+import com.typesafe.config.Config
+
+import scala.concurrent.duration._
+
+/**
+ * Settings for cluster distribution.
+ *
+ * @param clusterShardingSettings The cluster sharding settings.
+ * @param ensureActiveInterval The interval at which entities are ensured to be active.
+ */
+case class ClusterDistributionSettings(
+  clusterShardingSettings: ClusterShardingSettings,
+  ensureActiveInterval:    FiniteDuration
+)
+
+object ClusterDistributionSettings {
+  def apply(system: ActorSystem): ClusterDistributionSettings = {
+    val clusterShardingSettings = ClusterShardingSettings(system)
+    val config = system.settings.config.getConfig("lagom.persistence.cluster.distribution")
+    ClusterDistributionSettings(config, clusterShardingSettings)
+  }
+
+  def apply(config: Config, clusterShardingSettings: ClusterShardingSettings): ClusterDistributionSettings = {
+    val ensureActiveInterval = config.getDuration("ensure-active-interval", TimeUnit.MILLISECONDS).milliseconds
+    ClusterDistributionSettings(clusterShardingSettings, ensureActiveInterval)
+  }
+}
+
+object ClusterDistribution extends ExtensionId[ClusterDistribution] with ExtensionIdProvider {
+
+  override def createExtension(system: ExtendedActorSystem): ClusterDistribution =
+    new ClusterDistribution(system)
+
+  override def lookup = ClusterDistribution
+
+  override def get(system: ActorSystem): ClusterDistribution = super.get(system)
+
+  /**
+   * Sent to each entity in a cluster distribution to ensure it's active.
+   */
+  final case class EnsureActive(entityId: EntityId)
```

Comment:
You should add a proper serializer for this message to avoid Java serialization. 

If in Lagom it can be added to https://github.com/lagom/lagom/blob/master/persistence/src/main/scala/com/lightbend/lagom/internal/persistence/protobuf/PersistenceMessageSerializer.scala

If in Akka it can be added in https://github.com/akka/akka/blob/master/akka-cluster-sharding/src/main/scala/akka/cluster/sharding/protobuf/ClusterShardingMessageSerializer.scala